File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/StudentDataActivity.java
package com.marcinmoskala.kotlinapp;

import android.os.Bundle;
import android.widget.TextView;

import activitystarter.Arg;
import activitystarter.MakeActivityStarter;
import activitystarter.Optional;

@MakeActivityStarter
public class StudentDataActivity extends BaseActivity {

    private static int NO_ID = -1;

    @Arg @Optional String name = "No name provided";
    @Arg @Optional int id = NO_ID;
    @Arg char grade;
    @Arg boolean isPassing;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_data);
        TextView nameView = (TextView) findViewById(R.id.name_view);
        TextView idView = (TextView) findViewById(R.id.id_view);
        TextView gradeView = (TextView) findViewById(R.id.grade_view);
        TextView isPassingView = (TextView) findViewById(R.id.is_passing_view);

        nameView.setText("Name: "+name);
        idView.setText("Id: "+id);
        gradeView.setText("Grade: "+grade);
        isPassingView.setText("Passing status: "+isPassing);
    }
}

File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/StudentDataActivity.kt
package com.marcinmoskala.kotlinapp

import activitystarter.Arg
import activitystarter.MakeActivityStarter
import android.annotation.SuppressLint
import android.os.Bundle
import com.marcinmoskala.activitystarter.argExtra
import kotlinx.android.synthetic.main.activity_data.*

@MakeActivityStarter
class StudentDataActivity : BaseActivity() {

    @get:Arg(optional = true) var name: String by argExtra(defaultName)
    @get:Arg(optional = true) var id: Int by argExtra(defaultId)
    @get:Arg var grade: Char  by argExtra()
    @get:Arg var passing: Boolean  by argExtra()

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_data)
        nameView.text = "Name: $name"
        idView.text = "Id: $id"
        gradeView.text = "Grade: $grade"
        isPassingView.text = "Passing status: $passing"
    }

    companion object {
        private const val NO_ID = -1
        const val defaultName = "No name provided"
        const val defaultId = NO_ID
    }
}
